trio/_threads.py

```diff
@@ -266,9 +287,20 @@ async def to_thread_run_sync(  # type: ignore[misc]
 
     """
     await trio.lowlevel.checkpoint_if_cancelled()
-    if abandon_on_cancel is not None:
-        cancellable = abandon_on_cancel
-    abandon_on_cancel = bool(cancellable)  # raise early if cancellable.__bool__ raises
+    if cancellable is not None:
+        if abandon_on_cancel is not None:
+            raise ValueError(
+                "Cannot set `cancellable` and `abandon_on_cancel` simultaneously."
+            )
+        warnings.warn(
```

Comment:
Suggest you use `trio._deprecate.warn_deprecated()` which provides a standard message format and warning class (so users can suppress Trio deprecation warnings independently of others).